server/service/src/stock_line/update.rs

```diff
@@ -85,14 +95,38 @@ fn generate(
         batch,
         on_hold,
     }: UpdateStockLine,
-) -> StockLineRow {
-    existing.location_id = location_id.or(existing.location_id);
+) -> Result<(StockLineRow, Option<Vec<LocationMovementRow>>), UpdateStockLineError> {
+    existing.location_id = location_id.clone().or(existing.location_id);
     existing.batch = batch.or(existing.batch);
     existing.cost_price_per_pack = cost_price_per_pack.unwrap_or(existing.cost_price_per_pack);
     existing.sell_price_per_pack = sell_price_per_pack.unwrap_or(existing.sell_price_per_pack);
     existing.expiry_date = expiry_date.or(existing.expiry_date);
     existing.on_hold = on_hold.unwrap_or(existing.on_hold);
-    existing
+
+    let location_movements = if location_id != existing.location_id {
```

Comment:
the existing location_id is being set to the location_id in line 99.. so this conditional will never be true.
Just need to move the check above the setter in line 99